shopfloor/services/location_content_transfer.py

```diff
@@ -258,6 +280,54 @@ def _unreserve_other_lines(self, location, move_lines):
         unreserved_moves._do_unreserve()
         return (move_lines - lines_other_picking_types, unreserved_moves, None)
 
+    def _find_location_to_work_from(self):
+        next_picking = self.env["stock.picking"].search(
+            [
+                ("picking_type_id", "in", self.picking_types.ids),
+                ("state", "=", "assigned"),
+            ],
+            order="create_date",
+            limit=1,
+        )
+        return fields.first(next_picking.move_line_ids).location_id
```

Comment:
don't we filter on qty_done not full?

Reply:
as soon `shopfloor_reception` is merged, you'll be able to add `('progress', '<', 100)` in the domain.

Reply:
> don't we filter on qty_done not full?

Yes, added.